openid-caep-interoperability-profile-1_0.md

```diff
@@ -0,0 +1,193 @@
+# Introduction {#introduction}
+SSF and CAEP together enable improved session security outcomes. This specification defines the minimum required features from SSF and CAEP that an implementation MUST offer in order to be considered as an interoperable implementation. This document defines specific use cases. An implementation may support only a subset of the use cases defined herein, and SHALL be considered an interoperable implementation for the specific use-cases it supports. The following use-cases are considered as a part of this specification:
+
+Session Revocation
+: A SSF Transmitter or Receiver is able to respectively generate or respond to the CAEP session-revoked event
+
+Credential Change
+: A SSF Transmitter or Receiver is able to respectively generate or respond to the CAEP credential-change event
+
+# Common Requirements {#common-requirements}
+The following requirements are common across all use-cases defined in this document.
+
+## Transmitters {#common-transmitters}
+Transmitters MUST implement the following features:
+
+### Spec Version {#spec-version}
+The Transmitter Configuration Metadata MUST have a `spec_version` field, and its value MUST be `1_0-ID2` or greater
+
+### Delivery Method {#delivery-method}
+The Transmitter Configuration Metadata MUST include the `delivery_methods_supported` field.
+
+### JWKS URI {#jwks-uri}
+The Transmitter Configuration Metadata MUST include the `jwks_uri` field, and its value MUST provide the current signing key of the Transmitter.
+
+### Configuration Endpoint {#configuration-endpoint}
+The Transmitter Configuration Metadata MUST include the `configuration_endpoint` field. The specified endpoint MUST provide a way to Create a Stream.
+
+### Status Endpoint {#status-endpoint}
+The Transmitter Configuration Metadata MUST include the `status_endpoint` field. The specified endpoint MUST provide a way to Get and Update the Stream Status. The Transmitter MUST be able to pause and restart streams. For streams that are paused, the Transmitter MUST specify (offline) the resource constraints on how many events it can keep, or for how long. The way a Transmitter specifies this information is outside the scope of the SSF spec.
+
+### Verification Endpoint {#verification-endpoint}
+The Transmitter Configuration Metadata MUST include the `verification_endpoint` field. The specified endpoint MUST provide a way to request verification events to be sent.
+
+### Authorization Schemes
+The Transmitter Configuration Metadata MUST include the `authorization_schemes` field and its value MUST include the value
+
+~~~json
+{
+    "spec_urn": "urn:ietf:rfc:6749"
+}
+~~~
+
+### Streams {#common-stream-configuration}
+In all streams created by the Transmitter, the following MUST be true:
+
+#### Delivery {#common-delivery}
+The `delivery` field MUST be present in the Configuration of any Stream generated by the Transmitter, and its value MUST include the following:
+
+~~~ json
+{
+    "method" : "urn:ietf:rfc:8935"
```

Comment:
@atultulshi in your earlier comment you mentioned that PUSH is not a MUST have, then why do we suggest push to be a delivery method here?

Reply:
I agree. PUSH delivery gets set by the Receiver, not the Transmitter. So we should either say:
1. The Transmitter MUST include rfc:8936 in the delivery field
2. The Transmitter must be ready to accept a delivery specification of rfc:8935 from a Recevier

I actually think we should require both of those things.

Reply:
A stream can only be one of "push" or "poll", so I think we cannot have the language @FragLegs suggests. I'll make it so that the Transmitter must be ready to accept either.

Reply:
Excellent. The change you made is what I was trying to suggest. Sorry for my unclear language.